Source/Atomic/Resource/Image.cpp

```diff
@@ -2034,7 +2124,12 @@ unsigned char* Image::GetImageData(Deserializer& source, int& width, int& height
 
     SharedArrayPtr<unsigned char> buffer(new unsigned char[dataSize]);
     source.Read(buffer.Get(), dataSize);
-    return stbi_load_from_memory(buffer.Get(), dataSize, &width, &height, (int*)&components, 0);
+    // #623 BEGIN TODO: Crunch expects all 4 components. This isn't ideal, but since
+    // precompressed versions should always be loaded anyway it doesn't matter.
+    // Might need to handle differently per platform.
+    components = 4;
+    return stbi_load_from_memory(buffer.Get(), dataSize, &width, &height, nullptr, components);
+    // #623 END TODO: Crunch expects all 4 components. This isn't ideal, but since

```

Comment:
Add a bool parameter to GetImageData called forceComponents that defaults to false, if true it will force the components to whatever was passed in (by reference) otherwise will use component size of image and maintain compatible behavior?  If this sounds good, add some checks for 0 and > 4 components and log error return null.